### PR TITLE
feat: gate cli features

### LINKED ISSUE #39: Feature flag components
### Problem Statement

As a user that only needs access to the encode features of the crate it would be nice to not have to compile clap/etc dependencies that exist only for the CLI features

### Proposed Solution

Provide granular feature flags for the components. Included make serde derive an optional feature (derive is really heavy in compile times).

### Alternatives Considered

_No response_

### SPEC Compliance

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by jqnatividad**
for library users who do not want the dependencies required by the cli

## Linked Issue

Closes #39 

## Description

Gate deps required for cli tool behind cli feature.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Test coverage improvement

## Changes Made

Cargo.toml changes
- add a `cli` feature and group cli-related dependencies under it
- made the cli feature a `required-feature` for bin

lib.rs change
- gated tui module behind `cli` feature

## Testing

- [x] All existing tests pass
- [ ] Added new tests for changes
- [ ] Tests cover edge cases and spec compliance

## Pre-submission Checklist

- [x] My code follows the project's coding standards
- [ ] I have run code formatting/linting tools
- [ ] I have added tests that prove my fix/feature works
- [x] New and existing tests pass locally
- [ ] I have updated documentation if needed
- [ ] I have reviewed the [TOON specification](https://github.com/toon-format/spec) for relevant sections

## Breaking Changes

- [x] No breaking changes
- [ ] Breaking changes (describe migration path below)

### CODE REVIEW

**Review by shreyasbhat0 — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -22,26 +22,44 @@ path = "examples/main.rs"
 [[bin]]
 name = "toon"
 path = "src/cli/main.rs"
+required-features = ["cli"]
+
+[features]
+default = []
```
[the remark this answers is not in the document]
Comment (jqnatividad) [reply Cargo.toml]: Should have thought of that... Done!

**Comment by jqnatividad** [comment]
Hi @shreyasbhat0 ,
Went ahead and took the liberty of applying select clippy::pedantic lints and corrected some typos as well.

**Comment by jqnatividad** [comment]
FYI @shreyasbhat0 , I did `cargo fmt` but it changed 54 files so I didn't add it to the PR...
